File: src/Entity/Forum_interaction.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;


#[ORM\Entity]
class Forum_interaction
{

    #[ORM\Id]
    #[ORM\Column(type: "bigint")]
    private string $id;

    #[ORM\Column(type: "string")]
    private string $target_type;

    #[ORM\Column(type: "bigint")]
    private string $target_id;

    #[ORM\Column(type: "integer")]
    private int $user_id;

    #[ORM\Column(type: "string")]
    private string $interaction_type;

    #[ORM\Column(type: "datetime")]
    private \DateTimeInterface $created_at;

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getTarget_type()
    {
        return $this->target_type;
    }

    public function setTarget_type($value)
    {
        $this->target_type = $value;
    }

    public function getTarget_id()
    {
        return $this->target_id;
    }

    public function setTarget_id($value)
    {
        $this->target_id = $value;
    }

    public function getUser_id()
    {
        return $this->user_id;
    }

    public function setUser_id($value)
    {
        $this->user_id = $value;
    }

    public function getInteraction_type()
    {
        return $this->interaction_type;
    }

    public function setInteraction_type($value)
    {
        $this->interaction_type = $value;
    }

    public function getCreated_at()
    {
        return $this->created_at;
    }

    public function setCreated_at($value)
    {
        $this->created_at = $value;
    }
}

File: src/main/java/model/ForumInteraction.java
package model;

/**
 * FILE ROLE: Forum domain/data model definitions shared across layers.
 * FILE: Forum/src/main/java/model/ForumInteraction.java
 */

/**
 * LOGIC INDEX:
 * - Classes/Enums/Records: ForumInteraction
 * - File logic focus: Domain model and enum logic for forum entities.
 * - Feature coverage: State transport across UI/service/repo layers.
 * - Key methods/blocks: getId, setId, getTargetType, setTargetType, getTargetId, setTargetId, getUserId, setUserId, getInteractionType, setInteractionType
 */
import java.time.LocalDateTime;

/**
 * Data model for forum_interaction rows.
 */
public class ForumInteraction {
    private long id;
    private TargetType targetType;
    private long targetId;
    private int userId;
    private InteractionType interactionType;
    private LocalDateTime createdAt;

    public long getId() { return id; }
    public void setId(long id) { this.id = id; }

    public TargetType getTargetType() { return targetType; }
    public void setTargetType(TargetType targetType) { this.targetType = targetType; }

    public long getTargetId() { return targetId; }
    public void setTargetId(long targetId) { this.targetId = targetId; }

    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }

    public InteractionType getInteractionType() { return interactionType; }
    public void setInteractionType(InteractionType interactionType) { this.interactionType = interactionType; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}
